nomad/structs/structs.go

```diff
@@ -10612,8 +10612,19 @@ type DeploymentStatusUpdate struct {
 // RescheduleTracker encapsulates previous reschedule events
 type RescheduleTracker struct {
 	Events []*RescheduleEvent
+
+	// LastReschedule represents whether the most recent attempt to reschedule
+	// the allocation (if any) was successful
+	LastReschedule ResheduleTrackerAnnotation
 }
 
+type ResheduleTrackerAnnotation string
+
+const (
+	LastRescheduleSuccess       = "ok"
+	LastRescheduleFailedToPlace = "no placement"
```

Comment:
care to type these as Annotations?  sadly I can not recall right now under what circumstances this Really Matters

```suggestion
	LastRescheduleSuccess       ResheduleTrackerAnnotation = "ok"
	LastRescheduleFailedToPlace ResheduleTrackerAnnotation = "no placement"
```